### DISCUSSION TITLE
Cannot preview targets that are not a scheme

### DISCUSSION THREAD

**Discussion by amorino** [pointfreeco/isowords#100] [Q&A] [answered]
Hello, 

I'm having a problem previewing SwiftUI files that have not a scheme.

I tried deleted the DerivedData but it's a constant on any SwiftUI file that doesn't belong to a scheme. With files that belong to a scheme I don't have any problem, I post it here because I don't think is an issue with isowords, more of an issue with Xcode that's why I wanted to ask first here.

For example trying to preview `BottomMenu` `ComposableBottomMenu.swift` on scheme `isowords` show me the following error:

```
no such file or directory: '/Users/redacted/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/GameCore/Products/Debug-iphonesimulator/PackageFrameworks/BottomMenu.framework/BottomMenu'

----------------------------------------

LinkDylibError: Failed to build ComposableBottomMenu.swift

Linking failed: no such file or directory: '/Users/redacted/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/GameCore/Products/Debug-iphonesimulator/PackageFrameworks/BottomMenu.framework/BottomMenu'

clang: error: no such file or directory: '/Users/redacted/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/GameCore/Products/Debug-iphonesimulator/PackageFrameworks/BottomMenu.framework/BottomMenu'


==================================

|  BuildInvocationError
|  
|  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -F /Applications/Xcode.app/Contents/SharedFrameworks-iphonesimulator -target x86_64-apple-ios14.0-simulator -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk -L/Users/erickbazan/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/AppFeature/Products/Debug-iphonesimulator -L/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/lib -F/Users/erickbazan/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/AppFeature/Products/Debug-iphonesimulator -F/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -iframework /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/Library/Frameworks -iframework /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.4.sdk/Developer/Library/Frameworks -Xlinker -no_deduplicate -Xlinker -objc_abi_version -Xlinker 2 -fobjc-link-runtime -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -L/usr/lib/swift -dynamiclib /Users/erickbazan/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/AppFeature/Intermediates.noindex/isowords.build/Debug-iphonesimulator/BottomMenu.build/Objects-normal/x86_64/BottomMenu.2.preview-thunk.o -o /Users/erickbazan/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/AppFeature/Intermediates.noindex/isowords.build/Debug-iphonesimulator/BottomMenu.build/Objects-normal/x86_64/BottomMenu.2.preview-thunk.dylib /Users/erickbazan/Library/Developer/Xcode/DerivedData/isowords-gkbxccubvevjvobqspdrornsxwic/Build/Intermediates.noindex/Previews/AppFeature/Products/Debug-iphonesimulator/PackageFrameworks/BottomMenu.framework/BottomMenu -framework SwiftUI

```

**Comment by mbrandonw** [comment, marked as answer]
As far as I know you must have the scheme selected that holds the preview in order to run previews. 

So you gotta have the BottomMenu scheme selected to run the ComposableBottomMenu preview 😕. 

It does seem like as long as the selected scheme builds the preview you should be able to run it. May be worth filing a feedback with apple.

**Reply by amorino** [reply]
My bad, I didn't see that scheme, I was checking some other stuff and `Styleguide` doesn't have a scheme but has previews in some parts (`Buttons, Fonts`) and it failed with the same error. At the end, I think the `Styleguide` folder it's the only one I found with that problem. 

Anyways, I will fill feedback to Apple, I think it should detect the current scheme based on the target and code.

Thanks!

**Comment by Sajjon** [comment]
@amorino @mbrandonw you can also add the Package in question to any other scheme, e.g. I have an iOS App target called Zhip in the [Zhip app](https://github.com/OpenZesame/Zhip), it has lots of SPM Packages a la TCA, If I wanna preview lets say a package called `ConfirmNewPINFeature` I can just add that scheme to the _Zhip iOS Scheme_, under **_Build_**, I made sure the uncheck the checkbox `Archive`.

<img width="936" alt="Screenshot 2022-03-31 at 13 31 57" src="https://user-images.githubusercontent.com/864410/161046356-e44374d0-3b88-4c88-8239-6d6e2a9bdcef.png">

In fact, I also add the schemas for my test packages to **_Tests_** selection in the left menu of my _Zhip iOS Scheme_, so that I can run all tests using CMD U. This is convenient because then I no longer need to switch scheme while development, at all I think!

I do not know if this has any drawbacks? Maybe longer builds?

<img width="1147" alt="Screenshot 2022-03-31 at 13 35 37" src="https://user-images.githubusercontent.com/864410/161046402-c789874b-53c1-45b6-915b-8acc59618338.png">
